runtime/nvqir/CircuitSimulator.h

```diff
@@ -393,6 +403,12 @@ class CircuitSimulatorBase : public CircuitSimulator {
   /// is meant for subtypes to override
   virtual cudaq::State getStateData() { return {}; }
 
+  /// @brief Set the internal state representation. Each subtype will
+  /// override this, allowing for exposure at the user-API level via
+  /// the `cudaq::from_state` or a new `cudaq::set_state` function.
+  virtual void
```

Comment:
Set state data would become obsolete because all of this logic would now be wrapped up in the `addQubitsToState` function